src/engine/objects/robot/mechanisms/MechanismManager.ts

```diff
@@ -0,0 +1,107 @@
+import { SimMechanism } from "./SimMechanism";
+import { IRobotSpec, SensorSpec } from "../../../specs/RobotSpecs";
+//import { SimGripperMechanism } from "./SimGripperMechanism";
+import { EventRegistry } from "../../../EventRegistry";
+import { SimGripperMechanism } from "./SimGripperMechanism";
+
+/**
+ * Class representing a collection of robot mechanisms
+ *
+ * This class is not a SimObject, but instead a "manager" of SimObjects,
+ * in this instance, SimMechanism objects.
+ */
+export class MechanismManager {
+  private _mechanisms: Map<string, SimMechanism> = new Map<
+    string,
+    SimMechanism
+  >();
+
+  // map used to link robot channel numbers to a mechanism input
+  private _iomap: Map<
+    number,
+    { mechanismId: string; ioIdentifier: string }
+  > = new Map();
+
+  constructor(robotSpec: IRobotSpec, robotGuid: string) {
+    this._iomap = new Map();
+    this.configureMechanisms(robotSpec, robotGuid);
+  }
+
+  private configureMechanisms(robotSpec: IRobotSpec, robotGuid: string): void {
+    if (!robotSpec.mechanisms) {
+      return;
+    }
+
+    robotSpec.mechanisms.forEach((mechanismSpec) => {
+      let mechanism: SimMechanism;
+
+      if (mechanismSpec.type === "gripper-mechanism") {
```

Comment:
minor: the "gripper-mechanism" is the only mechanism as of now, right?  So maybe throw an error them `mechanismSpec.type` is set to something unknown?  Otherwise the rest of this function just operates on the mechanism as created on line 36.

Reply:
good idea, fixed